Source/Provers/SMTLib/SMTLibSolver.cs

```diff
@@ -16,43 +16,57 @@ public abstract class SMTLibSolver
 
   protected abstract void HandleError(string msg);
 
-  public void Ping()
+  public void Ping1()
   {
     Send("(get-info :name)");
   }
+  
+  public void Ping2()
+  {
+    Send("(get-info :version)");
+  }
```

Comment:
Not a fan of pong1 / pong2.  Can we give `ping` an argument saying what to send / look for?

Reply:
Maybe we can keep a single `Ping` but use `echo` with a counter instead of these `get-info` queries

Reply:
Nice, thanks for the pointer to the `echo` command in smt-lib ! it made the code nicer